test/integration/features/story_registry.feature

```diff
@@ -0,0 +1,50 @@
+Feature: Local image to image-registry to deployment
+
+    The user creates a local container image with an app. They then
+    push it to the openshift image-registry in their
+    project/namespace. They deploy and expose the app and check its
+    accessibility.
+
+    Scenario Outline: Start CRC
+        Given executing "crc setup" succeeds
+        When starting CRC with default bundle and hypervisor "<vm-driver>" succeeds
+        Then stdout should contain "CodeReady Containers instance is running"
+        And executing "eval $(crc oc-env)" succeeds
+        And executing "oc login --insecure-skip-tls-verify -u kubeadmin -p 9gZtH-EWZLT-mZNFP-bkKwm https://api.crc.testing:6443" succeeds
+        
+    @darwin
+        Examples:
+            | vm-driver  |
+            | hyperkit   |
+
+    @linux
+        Examples:
+            | vm-driver |
+            | libvirt   |
+
+    @darwin @linux @windows
+    Scenario: Create local image
+        Given executing "cd ../../../testdata" succeeds
+        When executing "sudo podman build -t hello:test ." succeeds
+        And executing "cd ../integration" succeeds
+        Then executing "sudo podman images" succeeds
+        And stdout should contain "localhost/hello"
+        
+    @darwin @linux @windows
+    Scenario: Push local image to OpenShift image registry
+        Given executing "oc new-project testproj" succeeds
+        When executing "sudo podman login -u kubeadmin -p $(oc whoami -t) default-route-openshift-image-registry.apps-crc.testing --tls-verify=false" succeeds
+        Then stdout should contain "Login Succeeded!"
+        When executing "sudo podman push hello:test default-route-openshift-image-registry.apps-crc.testing/testproj/hello:test --tls-verify=false" succeeds
+
+    @darwin @linux @windows
+    Scenario: Deploy the image
+        When executing "oc new-app testproj/hello:test" succeeds
+        Then with up to "3" retries with wait period of "1m" command "oc status" output should contain " - 1 pod"
```

Comment:
Instead of retry you should use `oc rollout status dc/hello` here `dc=deploymentconfig` which is a blocking call and wait till the resource are not available.

Reply:
Good tip! Now it's much neater.